src/traversals/dfs.jl

```diff
@@ -5,12 +5,37 @@
     is_cyclic(g)
 
 Return `true` if graph `g` contains a cycle.
-
+  
 ### Implementation Notes
-Uses DFS.
+The algorithm uses a DFS.
 """
 function is_cyclic end
-@traitfn is_cyclic(g::::(!IsDirected)) = ne(g) > 0
+@enum Vertex_state unvisited visited
+@traitfn function is_cyclic(g::AG::(!IsDirected)) where {T, AG<:AbstractGraph{T}}
+    vertex_state = Vector{Vertex_state}(undef, nv(g)) 
+    fill!(vertex_state,unvisited)
+    parent = zeros(nv(g))
+    for v in vertices(g)
+        if vertex_state[v] == unvisited
+            S = Vector{T}([v])
```

Comment:
```suggestion
            S = T[v]
```
The `T` specification is actually unnecessary here (but it may be easier to follow the code). But in any case, it is nicer than the `Vector{T}([v])` idiom which allocates two arrays (one for `[v]` and the other for `Vector{T}(...)` (unless the compiler is smart enough to remove one of those, which you should not rely on).

Reply:
Thank you! I didn't know this